core/src/test/java/com/linecorp/armeria/common/stream/StreamMessageTest.java

```diff
@@ -306,6 +308,82 @@ void writeToFile(@TempDir Path tempDir) throws IOException {
         }
     }
 
+    @Test
+    void noopSubscribe() {
+        final List<ByteBuf> bufs = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            bufs.add(Unpooled.wrappedBuffer(Integer.toString(i).getBytes()));
+        }
+        final HttpData[] httpData = bufs.stream().map(HttpData::wrap).toArray(HttpData[]::new);
+        final StreamMessage<HttpData> source = StreamMessage.of(httpData);
+        final CompletableFuture<Void> cf = source.subscribe();
+
+        await().untilAsserted(() -> assertThat(cf.isDone()).isTrue());
```

Comment:
Don't need to use `await()` here.
```suggestion
        assertThat(cf.join().isDone()).isTrue();
```